src/Integration/Suppression/SonarQubeIssuesProvider.cs

```diff
@@ -113,12 +114,27 @@ private string BuildModuleKey(string projectGuid)
             await SynchronizeSuppressedIssues();
         }
 
+        private Task DoInitialFetch()
+        {
+            // We might not have connected to the server at this point so if necessary
+            // wait before trying to fetch the issues
+            int retryCount = 0;
+            while (!this.sonarQubeService.IsConnected && retryCount < 30)
+            {
+                Thread.Sleep(1000);
```

Comment:
Isn't it better to do `await Task.Delay(1000)` here instead of `Thread.Sleep(1000)`? I think I remember to have read somewhere that we shouldn't mix both.

Reply:
_Task.Delay_ will start another task on another thread, whereas _Sleep_ will block the current thread. However, this code is already running in a separate task on a background thread so there doesn't seem to be much point in starting another task just to do the wait.